coalib/parsing/InvalidFilterException.py

```diff
@@ -1,2 +1,19 @@
-class InvalidFilterException(Exception):
-    pass
+class InvalidFilterException(BaseException):
+
+    def __init__(self, filter_name, available_filters):
+        self.filter = filter_name
+        self.available_filters = list(available_filters.split(', '))
+        if self.available_filters is None:
+            message = 'Invalid Filter'
+
+        if self.available_filters is not None:
+            message = 'Invalid filter. Available filters are : ' + \
+              str(self.available_filters)
+
+        else:
+            message = self.filter + \
+                      ' is an invalid filter. Available filters: ' + \
+                      str(self.available_filters)
+
+        super().__init__(message)
+        raise self
```

Comment:
never do this in an exception. The user shall raise exceptions himself, now the construction of this exception object always raises an exception.